yarn-project/pxe/src/storage/note_data_provider/note_data_provider.ts

```diff
@@ -284,20 +284,18 @@ export class NoteDataProvider implements DataProvider {
         const { data: nullifier, l2BlockNumber: blockNumber } = blockScopedNullifier;
         const noteIndex = await this.#nullifierToNoteId.getAsync(nullifier.toString());
         if (!noteIndex) {
-          continue;
+          throw new Error('Nullifier not found in removeNullifiedNotes');
```

Comment:
Error like these are a valuable feedback that we messed up somewhere.

Reply:
I'm a little bit torn on these errors. While I do agree this is great feedback for errors, you're potentially bricking someone's PXE if data got corrupted somehow, leaving them no choice but to nuke the db and sync from scratch.

Reply:
I see what you mean but it should not happen in the system that we would call `removeNullifiedNotes` on the note data provider with invalid nullifiers and at that point it would be hiding a serious bug and it would be over for the user's PXE anyway.

Reply:
Jan's right. It's different when e.g. processing messages that someone else sent, but at this stage we should defenitely have the nullif.